pages/public-sitemap.tsx

```diff
@@ -0,0 +1,153 @@
+import SitemapSection from "../components/sitemap/sitemapSection";
+import SitemapBlogSection from "../components/sitemap/sitemapBlogSection";
+import SitemapIntegrationSection from "../components/sitemap/sitemapIntegrationsSection";
+import { getBlogEntries } from "../utils/blogPosts";
+import SEO from "../components/seo";
+import paths from "../public/public-sitemap.json";
+import { badgeTypes } from "../data/blogTags";
+import { titleize } from "../utils/inflectors";
+import { SitemapDocsSection } from "../components/sitemap/sitemapDocsSection";
+
+export type SitemapItem = {
+  path: string;
+  name: string;
+};
+export interface Sitemap {
+  docs: Record<string, SitemapItem[]>;
+  legal: SitemapItem[];
+  solutions: SitemapItem[];
+  other: SitemapItem[];
+  integrations: {
+    sources: SitemapItem[];
+    destinations: SitemapItem[];
+    guides: SitemapItem[];
+  };
+  blog: Record<keyof typeof badgeTypes, SitemapItem[]>;
```

Comment:
The intent here is that blogs should be limited to our existing categories, docs should be dynamic to accommodate us adding or removing things as necessary in the future (think Profiles -> Records), and integrations should be fairly static in their categories.